src/services/validationService.js

```diff
@@ -0,0 +1,255 @@
+import lodash from 'lodash'
+
+import { validation as ValidationConstants } from '../constants'
+
+////// PRIVATE METHODS ///////
+
+/**
+ * Converts characters string to readable format
+ * Note: converts Hyphens to En Dashes, replaces one space with comma and space,
+ *       replaces letter `s` with `spaces` word
+ * @param {string} chars - characters to convert
+ * @returns {string} - converted string
+ * @example
+ * convertToLabel('a-z A-Z - _ *');
+ * // => 'a–z, A–Z, –, _, *'
+ */
+const convertToLabel = chars => {
+  return chars
+    .replace(/-/g, '–')
+    .replace(/\s/g, ', ')
+    .replace(/\bs\b/)
+}
+
+/**
+ * Converts characters string to valid RegExp string that will be placed into RegExp pattern
+ * @param {string} chars - characters to convert
+ * @returns {string} - converted string
+ * @example
+ * convertToPattern('a-z A-Z - _ *');
+ * // => 'a-zA-Z\-\_\*'
+ */
+const convertToPattern = chars => {
+  return chars
+    .split(' ')
+    .map(patternItem =>
+      patternItem.length === 1 ? '\\' + patternItem : patternItem
+    )
+    .join('')
+}
+
+/**
+ * Checks whether there is at least one failed validation rule.
+ * @returns {boolean} `true` in case there is at least one failed validation rule, or `false` otherwise.
+ */
+const hasInvalidRule = newRules => {
+  return lodash.some(newRules, ['isValid', false])
+}
+
+///////////////////////////////////////
```

Comment:
Maybe you wanted to add the comment that indicates that here are Public methods?